src/components/sidebar-navigation/SidebarNavigation.tsx

```diff
@@ -49,35 +53,57 @@ export const SidebarNavigation = forwardRef<
   });
 
   return (
-    <SidebarNavigationWrapper
-      aria-label={ariaLabel ?? defaultAriaLabel}
-      css={{
-        '--sidebar-navigation-width': isCollapsed
-          ? cssVar('sidebar-navigation-sizes-width-collapsed')
-          : cssVar('sidebar-navigation-sizes-width-expanded'),
-      }}
-      data-sidebar-collapsed={isCollapsed}
-      ref={ref}>
-      {children}
-    </SidebarNavigationWrapper>
+    <SidebarNavigationContainer>
+      <SidebarNavigationWrapper aria-label={ariaLabel ?? defaultAriaLabel} ref={ref}>
+        {children}
+      </SidebarNavigationWrapper>
+    </SidebarNavigationContainer>
   );
 });
 
 SidebarNavigation.displayName = 'SidebarNavigation';
 
-const SidebarNavigationWrapper = styled.nav`
-  box-sizing: content-box;
-  width: var(--sidebar-navigation-width);
-  background-color: ${cssVar('color-surface-canvas-default')};
-  border-right: ${cssVar('border-width-default')} solid ${cssVar('color-border-weak')};
+const SidebarNavigationContainer = styled.div`
+  grid-area: sidebar;
+  position: relative;
 
-  overflow: hidden;
+  width: calc(var(--sidebar-navigation-container-width) + ${cssVar('border-width-default')});
 
-  padding-top: ${cssVar('dimension-space-250')};
+  --sidebar-navigation-container-width: ${cssVar('sidebar-navigation-sizes-width-closed')};
+
+  [data-sidebar-docked='true'] & {
+    --sidebar-navigation-container-width: ${cssVar('sidebar-navigation-sizes-width-open')};
+  }
+`;
+SidebarNavigationContainer.displayName = 'SidebarNavigationContainer';
+
+const SidebarNavigationWrapper = styled.nav`
+  position: absolute;
+  top: 0;
+  bottom: 0;
 
   display: flex;
   flex-direction: column;
+  box-sizing: content-box;
+  overflow: hidden;
+
+  padding-top: ${cssVar('dimension-space-250')};
+  border-right: ${cssVar('border-width-default')} solid ${cssVar('color-border-weak')};
+  background-color: ${cssVar('color-surface-canvas-default')};
 
   transition: width 0.1s;
+
+  width: var(--sidebar-navigation-width);
+
+  --sidebar-navigation-width: ${cssVar('sidebar-navigation-sizes-width-open')};
+
+  // hover and focus-within pilotes the open state of the sidebar
+  [data-sidebar-docked='false'] &:not(:hover, :focus-within) {
+    --sidebar-navigation-width: ${cssVar('sidebar-navigation-sizes-width-closed')};
+  }
+
+  [data-sidebar-docked='true'] &:is(:hover, :focus-within) {
+    box-shadow: ${cssVar('box-shadow-x-large')};
+  }
```

Comment:
should be `data-sidebar-docked='false'`

Reply:
Indeed, I caught it while working on some tests and fixed with my last commit